### PR TITLE
Draft HTTP Response model

### LINKED ISSUE #5: HTTP model for Response
Create `zio.http.Response` that will model the HTTP response.

### PR DESCRIPTION
**Pull request by co-category**
Fixes #5

### CODE REVIEW

File: core/src/main/scala/zio/http/Response.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package zio.http
+
+import zio.http.model.Header
+
+final case class Response[T](
+  headers: List[Header],
+  status: StatusCode,
+  cookies: List[Cookie],
+  contentType: String,
```
Comment (ioleo) [inline core/src/main/scala/zio/http/Response.scala]: We should use `ContentType` from #12

File: core/src/main/scala/zio/http/Response.scala [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package zio.http
+
+import zio.http.model.Header
```
Comment (ioleo) [inline core/src/main/scala/zio/http/Response.scala]: Perhaps you could move the header one package up, currently we're stuffing everything in `zio.http`.

File: core/src/main/scala/zio/http/Response.scala (hunk 1 above)
Comment (co-category) [reply core/src/main/scala/zio/http/Response.scala]: Sure. I was not sure if that would be merged and left it as a String until resolved. I've added that definition to my PR as it looks as though #12  is still not merged.

File: core/src/main/scala/zio/http/Response.scala (hunk 2 above)
Comment (co-category) [reply core/src/main/scala/zio/http/Response.scala]: I've refactored the model structure. WDYT?

**Comment by vitaly-pushkar** [comment]
Could you link the issue https://github.com/zio/zio-http/issues/5 in the PR description so that it gets linked there, please?

File: core/src/main/scala/zio/http/Response.scala (hunk 1 above)
Comment (ioleo) [reply core/src/main/scala/zio/http/Response.scala]: @co-category Its merged now.

File: core/src/main/scala/zio/http/Response.scala (hunk 2 above)
Comment (ioleo) [reply core/src/main/scala/zio/http/Response.scala]: @co-category please keep the `zio.http` prefix, the `zio` main package is reserved to zio core. I think we could move the data structures to `zio.http.model`.
Also please rebase to master, #12 was merged now.

Comment (co-category) [reply core/src/main/scala/zio/http/Response.scala]: @ioleo sorry, misunderstood your original comment. Moved all the current data structures to `zio.http.model` and moved `Header` to `zio.http`. Also have rebased @vitaly-pushkar changes.

**Comment by co-category** [comment]
@ioleo I noticed the headers are not being generated for new files or existing one without a header. Does the zio team use `sbt-header`?

**Comment by ioleo** [comment]
@co-category Yes, we definately should add `sbt-header`. Regarding the PR, sorry for the confusion, if you'd be so kind and move `Headers` back to `zio.http.model` (such that all our data structures modeling the HTTP protocol are in one package) and it's good to go. :pray:

**Comment by co-category** [comment]
@ioleo moved `Header` also to the model package. Also added `sbt-header` and the associated header definition.

File: build.sbt [hunk 3]
```diff
@@ -55,3 +57,25 @@ lazy val core = project
       "dev.zio" %% "zio-nio" % zioNioVersion
     )
   )
+  .settings(
```
Comment (mschuwalow) [inline build.sbt]: If we are using sbt-header do we also want to run headerCheck in CI?

Comment (co-category) [reply build.sbt]: good point, we could just add the `headerCreate` to the `check` task.

Comment (co-category) [reply build.sbt]: unfortunately `sbt-header` doesn't work well on multi-module projects (see [here](https://github.com/sbt/sbt-header/issues/153#issuecomment-427182112))

I will take out out the `sbt-header` integration for now and maybe look to raise a PR to fix the issue in `sbt-header` in the future.

**Comment by ioleo** [comment]
@co-category Thank you for contribution!

**Pull request merged**